homeassistant/components/fronius/sensor.py

```diff
@@ -1,348 +1,732 @@
+    coordinator: FroniusCoordinatorBase
+    entity_descriptions: list[SensorEntityDescription]
+    _entity_id_prefix: str
 
     def __init__(
-        self, bridge: Fronius, name: str, device: int, add_entities: AddEntitiesCallback
+        self,
+        coordinator: FroniusCoordinatorBase,
+        key: str,
+        solar_net_id: str,
     ) -> None:
-        """Initialize the sensor."""
-        self.bridge = bridge
-        self._name = name
-        self._device = device
-        self._fetched: dict[str, Any] = {}
-        self._available = True
-
-        self.sensors: set[str] = set()
-        self._registered_sensors: set[SensorEntity] = set()
-        self._add_entities = add_entities
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def data(self):
-        """Return the state attributes."""
-        return self._fetched
-
-    @property
-    def available(self):
-        """Whether the fronius device is active."""
-        return self._available
-
-    async def async_update(self):
-        """Retrieve and update latest state."""
-        try:
-            values = await self._update()
-        except FroniusError as err:
-            # fronius devices are often powered by self-produced solar energy
-            # and henced turned off at night.
-            # Therefore we will not print multiple errors when connection fails
-            if self._available:
-                self._available = False
-                _LOGGER.error("Failed to update: %s", err)
-            return
-
-        self._available = True  # reset connection failure
-
-        attributes = self._fetched
-        # Copy data of current fronius device
-        for key, entry in values.items():
-            # If the data is directly a sensor
-            if "value" in entry:
-                attributes[key] = entry
-        self._fetched = attributes
-
-        # Add discovered value fields as sensors
-        # because some fields are only sent temporarily
-        new_sensors = []
-        for key in attributes:
-            if key not in self.sensors:
-                self.sensors.add(key)
-                _LOGGER.info("Discovered %s, adding as sensor", key)
-                new_sensors.append(FroniusTemplateSensor(self, key))
-        self._add_entities(new_sensors, True)
-
-        # Schedule an update for all included sensors
-        for sensor in self._registered_sensors:
-            sensor.async_schedule_update_ha_state(True)
-
-    async def _update(self) -> dict:
-        """Return values of interest."""
+        """Set up an individual Fronius meter sensor."""
+        super().__init__(coordinator)
+        self.entity_description = next(
+            desc for desc in self.entity_descriptions if desc.key == key
+        )
+        self.entity_id = f"{SENSOR_DOMAIN}.{key}_{DOMAIN}_{self._entity_id_prefix}_{coordinator.solar_net.host}"
```

Comment:
We recommend not setting an entity id explicitly when setting unique id since the user can change both name and entity id easily via the frontend.

Regardless, this is not the PR to change this.

Reply:
This is added to keep backwards compatibility to the current Fronius implementation. 

As soon as the entities hit the entity_registry this setting isn't used anymore, is it?

I'd like to keep it for a couple of releases to not break automations and energy panel assignments for upgrading users. 

After that period I wanted to change it to a more practical format for new users. If you prefer it can be completely removed as well then.

Reply:
Why can't we just set the name as we did before?

Reply:
I'm confused now.
- Do you mean before as in before 5ce4919 and reintroduce the breaking entity id changes?
- Or do you mean to use the naming schema as in the current integration. This would yield names like "current_ac_phase_3_fronius_meter_0_http_10_1_2_6" which in my opinion isn't particulary user friendly.

Reply:
The latter yes. If we want to change the naming schema it should be a separate PR.

Reply:
Shall I remove the SensorEntityDescriptions too then? Or just the name key?

Reply:
We can keep the descriptions. Just don't add any more features or other behavior changes that are user facing.

Reply:
I reverted to the current naming schema according to 
https://github.com/home-assistant/core/blob/674993073673039e730db3efc56c87b727f07cae/homeassistant/components/fronius/sensor.py#L313
and its `parent`
https://github.com/home-assistant/core/blob/674993073673039e730db3efc56c87b727f07cae/homeassistant/components/fronius/sensor.py#L137